### ISSUE TITLE
Is it safe to treat library functions as pure?

### ISSUE THREAD

**Issue by paluh** [oreshinya/purescript-simple-jwt#3]
Thanks once again for building this library.

It seems that decoding and encoding could be treated as pure. What do you think?

**Comment by paluh** [comment]
@oreshinya Do you think that it could be a security issue if we lift `bytestring` encoding / decoding errors to the PS level and treat high level library functions as `pure` ?

**Comment by oreshinya** [comment]
@paluh 
Sorry for the late response.

You have a point, so I will implement and release new version in a few days.

**Comment by paluh** [comment]
> Sorry for the late response.

No problem at all - thanks a lot for responding!

> You have a point

To be honest we have started already using it in a somewhat "unsafe manner":  https://github.com/purescript-webrow/webrow/blob/master/src/WebRow/Crypto/Jwt.purs#L20 ;-)

Thanks again for building a lot of useful libs!

**Comment by oreshinya** [comment]
After some thought, I decided not to change the interface.

There are cases that  `Effect Jwt` is more user friendly than `Either JwtError Jwt` as `encode` result.
(For example, the case that errors of native effects are caught on any framework side, and so on.)

So whether pure function is easier to use depends on the situation.

So I decided not to change the interface  as the stance of this library.

Thank you.
